### PR TITLE
Added cpuhotplug with stress tests and iterations

### PR DESCRIPTION
**Pull request by sathnaga**
Added cpuhotplug with avocado test(stress) in background
and test to run multiple iterations.

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by sathnaga** [comment]
```
 (1/3) guest_import.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-qemu.unattended_install.import.import.default_install.aio_native: PASS (41.52 s)
 (2/3) guest_cpu.hotplug.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-libvirt.libvirt_vcpu_plug_unplug.positive_test.vcpu_set.live.with_stress: PASS (666.74 s)
 (3/3) guest_cpu.hotplug.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-libvirt.libvirt_vcpu_plug_unplug.positive_test.vcpu_set.live.with_iteration: PASS (176.98 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 891.50 s
```

**Comment by sathnaga** [comment]
```
 (1/3) guest_import.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-qemu.unattended_install.import.import.default_install.aio_native: PASS (41.52 s)
 (2/3) guest_cpu.hotplug.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-libvirt.libvirt_vcpu_plug_unplug.positive_test.vcpu_set.live.with_stress: PASS (666.74 s)
 (3/3) guest_cpu.hotplug.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-libvirt.libvirt_vcpu_plug_unplug.positive_test.vcpu_set.live.with_iteration: PASS (176.98 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 891.50 s
```

File: libvirt/tests/src/libvirt_vcpu_plug_unplug.py [hunk 1]
```diff
@@ -434,6 +437,25 @@ def run(test, params, env):
         if 'power' not in cpu_util.get_cpu_arch():
             vmxml.set_pm_suspend(vm_name, "yes", "yes")
         vm.start()
+        if with_stress:
+            testlist = []
+            avocadotest = params.get("avocadotest", "cpu/ebizzy.py")
+            avocadomux = params.get("avocadomux", "")
+            avocadotestargs = params.get("avocadotestargs", "")
+            avocadotimeout = int(params.get("avocadotimeout", 3600))
+            testrepo = "https://github.com/avocado-framework-tests/avocado-misc-tests.git"
```
[the remark this answers is not in the document]
Comment (sathnaga) [reply libvirt/tests/src/libvirt_vcpu_plug_unplug.py]: @balamuruhans will validate the option of extending this in to library and call it from here

File: libvirt/tests/cfg/libvirt_vcpu_plug_unplug.cfg [hunk 2]
```diff
@@ -62,6 +62,14 @@
                                     pin_after_unplug = "yes"
                                     pin_vcpu = "0"
                                     pin_cpu_list = "x"
+                        - with_stress:
+                            only live
+                            test_itr = 4
+                            run_stress = "yes"
+                            avocadotest = "cpu/ebizzy.py"
```
Comment (chunfuwen) [inline libvirt/tests/cfg/libvirt_vcpu_plug_unplug.cfg]: avocadotest to avocado_test

File: libvirt/tests/src/libvirt_vcpu_plug_unplug.py [hunk 3]
```diff
@@ -434,6 +437,25 @@ def run(test, params, env):
         if 'power' not in cpu_util.get_cpu_arch():
             vmxml.set_pm_suspend(vm_name, "yes", "yes")
         vm.start()
+        if with_stress:
+            testlist = []
+            avocadotest = params.get("avocadotest", "cpu/ebizzy.py")
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_vcpu_plug_unplug.py]: avocadotest to avocado_test ?

File: libvirt/tests/src/libvirt_vcpu_plug_unplug.py (hunk 1 above)
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_vcpu_plug_unplug.py]: Maybe not hardcode the url ,and can be put into cfg file as parameter

**Comment by sathnaga** [comment]
Dependancy PR: https://github.com/avocado-framework/avocado-vt/pull/1157

**Comment by sathnaga** [comment]
```
 (1/3) guest_sanity.hotplug.import.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-qemu.unattended_install.import.import.default_install.aio_native: PASS (43.61 s)
 (2/3) guest_sanity.hotplug.cpu.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-libvirt.libvirt_vcpu_plug_unplug.positive_test.vcpu_set.live.with_stress: PASS (144.68 s)
 (3/3) guest_sanity.hotplug.delete_guest.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.26.ppc64le.io-github-autotest-libvirt.remove_guest.without_disk: PASS (1.16 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 196.21 s
```

File: libvirt/tests/src/libvirt_vcpu_plug_unplug.py [hunk 4]
```diff
@@ -451,182 +457,185 @@ def run(test, params, env):
...
+                    expect_vcpu_num[2] = vcpu_plug_num
+                elif setvcpu_option == "--guest":
+                    # vcpuset '--guest' only affect vcpu number in guest
+                    expect_vcpu_num[4] = vcpu_plug_num
+                else:
+                    expect_vcpu_num[3] = vcpu_plug_num
+                    expect_vcpu_num[4] = vcpu_plug_num
+                    if not status_error:
+                        if not online_new_vcpu(vm, vcpu_plug_num):
+                            raise error.TestFail("Fail to enable new added cpu")
```
Comment (balamuruhans) [inline libvirt/tests/src/libvirt_vcpu_plug_unplug.py]: lets use `test.fail()` and accordingly in other places

Comment (sathnaga) [reply libvirt/tests/src/libvirt_vcpu_plug_unplug.py]: @balamuruhans Sure, will raise separate PR, can we merge this one for addressing the same, Thanks :-)

Comment (balamuruhans) [reply libvirt/tests/src/libvirt_vcpu_plug_unplug.py]: okay.

**Pull request merged**